### DISCUSSION TITLE
[@types/aws-lambda] Wrong SQS handler response type for aws-lambda

### DISCUSSION THREAD

**Discussion by jorgevrgs** [DefinitelyTyped/DefinitelyTyped#68493] [Issues with a @types package] [answered]
Based on this type:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/trigger/sqs.d.ts#L52

The response of the lambda should be:

```typescript
{
  batchItemFailures: [{itemIdentifier: 'id1'},{itemIdentifier: 'id2'}]
}
```

However, based on this docs https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#services-sqs-batchfailurereporting the correct response is:

```typescript
{
  statusCode: 200,
  body: JSON.stringify({batchItemFailures: [{itemIdentifier: 'id1'},{itemIdentifier: 'id2'}]})
}
```

Is that correct?

[a comment by an automated account is not shown]

**Reply by simonbuchan** [reply, marked as answer]
That looks like documentation error, as the non-JavaScript examples all just directly return the value. Perhaps both work, though?

**Reply by jorgevrgs** [reply]
does it seems to be more like a `ReportBatchItemFailures` configuration? I found the examples with the response:

https://github.com/aws-samples/serverless-snippets/blob/main/lambda-function-sqs-report-batch-item-failures/example.js

**Reply by jorgevrgs** [reply]
You're right that all the other examples are according to the provided types. I'll double check the response.

**Reply by simonbuchan** [reply]
Seems hard to test, unfortunately

**Reply by jorgevrgs** [reply]
I tested out with `FunctionResponseTypes: ReportBatchItemFailures`:
https://docs.aws.amazon.com/lambda/latest/api/API_UpdateEventSourceMapping.html#API_UpdateEventSourceMapping_RequestSyntax

So for this option the type is correct and the  aws docs are outdated.

Thanks for your quick reply. It helped me a lot.

**Reply by simonbuchan** [reply]
No problem! Unfortunately it's somewhat common that the docs are outright wrong, which is unfortunate.
